### ISSUE TITLE
Fetch directly from crates.io if given a crate name

### ISSUE THREAD

**Issue by thesamesam** [projg2/pycargoebuild#14]
It'd be nice if pycargoebuild could fetch directly from crates.io when given a crate name.

This would be useful for two reasons:
1. Laziness: right now, I have to grab whatever the ebuild would fetch, unpack it, then run `pycargoebuild` inside. Lots of effort.
2. Required for some software: pkgcraft-c's [Cargo.toml](https://github.com/pkgcraft/pkgcraft/blob/main/crates/pkgcraft-c/Cargo.toml#L22) refers to pkgcraft via a path (`../pkgcraft`) which neither `pycargoebuild` nor `cargo-ebuild` handle properly. Neither end up listing `pkgcraft` as a dependency. The version downloaded from crates.io _does_ list it properly because `cargo` mangles it first on upload (or crates.io mangles it before publishing, whatever).

**Comment by gyakovlev** [comment]
You don’t need to pre-fetch generally.
Just 
`cd pkgcraft-c && cargo_src_compile`
You dont want to use crate tarball for workspace/vendored source like this, that’s why it’s omitted from CRATES

**Comment by mgorny** [comment]
I suppose this could be done in two steps:

1. Support specifying URL instead of directory, and unpack it to tmpdir.
2. Support specifying crate name instead of URL.

Am I missing something?

**Comment by thesamesam** [comment]
No, I think you're spot on. Both would be helpful and then together, it solves my bother.

**Comment by mgorny** [comment]
So after spending >1 hour on implementing this, I'm stuck because crates don't have `Cargo.lock`.

**Comment by Aromate** [comment]
> So after spending >1 hour on implementing this, I'm stuck because crates don't have `Cargo.lock`.

Can use `cargo generate-lockfile` create the `Cargo.lock`
